### PR TITLE
[Merge on evening 5-May] New blog: OptaPlanner enlightenment starts in the logs

### PR DESCRIPTION
**Pull request by ge0ffrey**
Related: https://github.com/kiegroup/optaplanner/pull/1296

Wait until merging until 5-May, so it shows up on the kie blog on 6-May.

### CODE REVIEW

File: blog/2021-05-06-ThePathToOptaPlannerEnlightenmentStartsInTheLog.adoc [hunk 1]
```diff
@@ -0,0 +1,311 @@
...
+If so, keep reading.
+
+_The Age of Enlightenment centered on the use of reason and the evidence of the senses.
+Similarly, OptaPlanner enlightenment starts with the use of reason and the evidence in the logs._
+
+OK. That's enough philosophy for this article... It's time to open the box.
+Take a look at the log for https://youtu.be/7IuOA9n6kh0[the school timetabling AI quickstart]:
+
+[source]
+----
```
Comment (yurloc) [inline blog/2021-05-06-ThePathToOptaPlannerEnlightenmentStartsInTheLog.adoc:20]: Preferably, use a [literal block](https://docs.asciidoctor.org/asciidoc/latest/verbatim/literal-blocks/#delimited-literal-block) for log outputs. [Source](https://docs.asciidoctor.org/asciidoc/latest/verbatim/source-blocks/) is for syntax highlighting.

The consequence of using `[source]` for log output in this PR is that some blocks are rendered without syntax highlighting and some blocks are detected as CSS code and are highlighted in a funny way.

File: blog/2021-05-06-ThePathToOptaPlannerEnlightenmentStartsInTheLog.adoc [hunk 2]
```diff
@@ -0,0 +1,311 @@
...
+* The Solver initialization took `196` milliseconds.
+* The Construction Heuristic took `330 - 196` = `134` milliseconds.
+** For big datasets, if the CH takes more than one minute,
+it's worth investigating why, even if the solver runs for hours.
+* The Local Search took `30000 - 330` = `29670` milliseconds.
+** LS gets the lion's share of the time, as it should be.
+
+== Score quality
+
+The log also revels the solution quality after each algorithm:
```
Comment (yurloc) [inline blog/2021-05-06-ThePathToOptaPlannerEnlightenmentStartsInTheLog.adoc:90]:
```suggestion
The log also reveals the solution quality after each algorithm:
```

File: blog/2021-05-06-ThePathToOptaPlannerEnlightenmentStartsInTheLog.adoc [hunk 3]
```diff
@@ -0,0 +1,311 @@
...
+           // Evaluate move
+       }
+       // Take step
+   }
+----
+
+The outer, step loop executes the best move found by the inner, move loop.
+Of course, this is a gross simplification:
+there are dozens of orthogonal AI subsystems on top of it.
+It's only the tip of the tip of the iceberg. But it's an honest simplification.
```
Comment (yurloc) [inline blog/2021-05-06-ThePathToOptaPlannerEnlightenmentStartsInTheLog.adoc:164]: Double "tip of the" intentional?

File: blog/2021-05-06-ThePathToOptaPlannerEnlightenmentStartsInTheLog.adoc [hunk 4]
```diff
@@ -0,0 +1,311 @@
...
+
+The `INFO` log shows how many of these outer loop iteration both phases did:
```
Comment (yurloc) [inline blog/2021-05-06-ThePathToOptaPlannerEnlightenmentStartsInTheLog.adoc:166]:
```suggestion
The `INFO` log shows how many of these outer loop iterations both phases did:
```

**Review by yurloc — CHANGES_REQUESTED** [review]

File: blog/2021-05-06-ThePathToOptaPlannerEnlightenmentStartsInTheLog.adoc (hunk 1 above)
Comment (ge0ffrey) [reply blog/2021-05-06-ThePathToOptaPlannerEnlightenmentStartsInTheLog.adoc:20]: This is exactly what I was looking for! Thank you!

**Pull request merged**